any_parser/any_parser.py

```diff
@@ -3,254 +3,209 @@
 
+    @handle_parsing
     def parse(
         self,
-        file_path: str,
-        extract_args: Optional[Dict] = None,
-    ) -> Tuple[str, str]:
-        """Extract full content from a file in real-time.
-
-        Args:
-            file_path (str): The path to the file to be parsed.
-            extract_args (Optional[Dict]): Additional extraction arguments added
-                to the prompt.
-
-        Returns:
-            tuple(str, str): The extracted data and the time taken.
-        """
-
-        response, info = self._get_sync_response(
-            self._sync_parse_url,
-            file_path,
+        file_path=None,
+        file_content=None,
+        file_type=None,
+        extract_args=None,
+    ):
+        """Extract full content from a file in real-time."""
+        response, info = self._sync_parser.get_sync_response(
+            self._sync_parser._sync_parse_url,
+            file_content=file_content,  # type: ignore
+            file_type=file_type,  # type: ignore
             extract_args=extract_args,
         )
+
         if response is None:
             return info, ""
 
         try:
             response_data = response.json()
-            response_list = []
-            for text in response_data["markdown"]:
-                response_list.append(text)
-            markdown_text = "\n".join(response_list)
-            return (
-                markdown_text,
-                f"Time Elapsed: {info}",
-            )
+            result = "\n".join(
+                response_data["markdown"]
+            )  # Using direct extraction instead of extract_key
+            return result, f"Time Elapsed: {info}"
         except json.JSONDecodeError:
             return f"Error: Invalid JSON response: {response.text}", ""
 
+    @handle_parsing
     def extract_pii(
         self,
-        file_path: str,
-    ) -> Tuple[str, str]:
-        """Extract personally identifiable information (PII) from a file in real-time.
-
-        Args:
-            file_path (str): The path to the file to be parsed.
-        Returns:
-            tuple(str, str): The extracted data and the time taken.
-        """
-        response, info = self._get_sync_response(
-            self._sync_extract_pii,
-            file_path,
+        file_path=None,
+        file_content=None,
+        file_type=None,
+    ):
+        """Extract PII from a file in real-time."""
+        response, info = self._sync_parser.get_sync_response(
+            self._sync_parser._sync_extract_pii,
+            file_content=file_content,  # type: ignore
+            file_type=file_type,  # type: ignore
             extract_args=None,
         )
+
         if response is None:
             return info, ""
 
         try:
             response_data = response.json()
-            return (
-                response_data["pii_extraction"],
-                f"Time Elapsed: {info}",
-            )
+            result = response_data["pii_extraction"]
```

Comment:
I was talking with @Sdddell offline that it can be confusing to use "markdown", "pii_extraction" as the response_data, we should make a TODO in the future to unify our backend to has a key for all response.

Also, it is a very bad idea to access a response_data through dict. After the above change is done, I suggest you to have a dataclass to always convert the response_data into a class and access through variable instead of key. This is dramatically improve the code readability and robustness to avoid suddenly key not in dict issue. TL'DR to access a data structure as dict is the worse idea and you should always try to serialized and deserialized.